Individualize RSpec rake tasks

miguelsorianod opened the pull request:
I thought it would be a good idea to be able to execute RSpec tests inividually or by type of test (unit, integration, ...) via Rake from within the development environment. I have created specific Rake tasks to execute them.

Let me know if you see any problems. In each commit I have written the motivations of the changes associated to it.

Please tell me if this changes could cause problems in the current integration/testing systems  to see if I may be generating some incompatibility.

Thank you.

unleashed commented on Rakefile:
```diff
@@ -65,8 +65,48 @@ if saas?
+      end
+
+      desc 'Run RSpec test/s specified by input file pattern'
+      RSpec::Core::RakeTask.new(:specific, :test_name) do | t, task_args |
```
style `|arg1, arg2|` - also, you could use `task` instead of `t` just like in the previous examples

unleashed commented on Rakefile:
```diff
@@ -102,8 +102,8 @@ if saas?
       desc 'Run RSpec test/s specified by input file pattern'
-      RSpec::Core::RakeTask.new(:specific, :test_name) do | t, task_args |
-        t.pattern = "#{task_args[:test_name]}"
+      RSpec::Core::RakeTask.new(:specific, :test_name) do | task, task_args |
```
ah, fixed the name here - the style issue remains, and you should squash this commit into the one that introduces these changes.

unleashed commented on README.md:
```diff
@@ -98,6 +98,37 @@ variables for the test scripts like so:
+> `$ bundle exec rake`
+
+Another alternative to execute all tests from within the development environment
+where it is not needed to manually start the services before is:
```
this is not correct: `it is not needed` should be replaced with `you should not` (ie. it might work only out of chance and it is likely to produce problems) - and maybe mention how to stop the services in case you had them running.

unleashed commented on README.md:
```diff
@@ -98,6 +98,37 @@ variables for the test scripts like so:
+
+For more information on accepted commands for testing with Rake you can execute:
+
+> `$bundle exec rake -T`
```
typo: place a space in between the prompt and `bundle`.

unleashed commented on README.md:
```diff
@@ -98,6 +98,37 @@ variables for the test scripts like so:
+
+where in this last case the [ ] characters must be literally placed.
+
+For more information on accepted commands for testing with Rake you can execute:
```
you could also document how to use rspec directly with `bundle exec rspec <parameters> <spec_file>` to run a specs file.

unleashed commented on Rakefile:
```diff
@@ -79,31 +79,37 @@ if saas?
       desc 'Run RSpec test/s specified by input file pattern'
       RSpec::Core::RakeTask.new(:specific, :test_name) do | task, task_args |
         task.pattern = "#{task_args[:test_name]}"
+        task.rspec_opts = '--require=spec_helper'
```
these changes should also be squashed in the commit introducing them - perhaps add a comment in the first line doing this to explain why it is needed.

unleashed commented on Rakefile:
```diff
@@ -65,8 +65,48 @@ if saas?
+      desc 'Run RSpec use_cases tests'
+      RSpec::Core::RakeTask.new(:use_cases) do |task|
+        task.pattern = 'spec/use_cases/**/*_spec.rb'
+      end
```
Nice job - however, when I look at all these rules there is a DRY flag being raised: they all do the same except for a very small thing. This would be a bit of advanced stuff, but can we come up with a way to have Ruby write all these for us?

unleashed commented:
@miguelsorianod take a look at the "outdated" comments too, as I was writing some of them on invididual diffs that later got further modified.

davidor commented on README.md:
```diff
@@ -102,6 +102,44 @@ variables for the test scripts like so:
+Another alternative to execute all tests from within the development environment
+where you should not need to manually start/stop the services before is:
+
+> `$ script/test`
```
I think this should be the first one to appear.

The documentation for running the tests has become quite complex with all the possible options and I think that the "normal way" (`script/test`) should be more visible.

miguelsorianod commented:
I have updated the README.md in order to have the basic steps needed to execute all the tests to be listed first in the testing section and then I added a subsection in the readme (running tests (advanced)) on how to execute the tests in more advanced ways

eguzki commented:
bors r=@unleashed,@davidor,@eguzki
